src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/CompileExpressionsTests.cs

```diff
@@ -705,5 +706,199 @@ .maxstack  2
 }");
                 });
         }
+
+        [Fact]
+        public void FileLocalType_01()
+        {
+            var source =
+@"file class C
+{
+    public static int X = 42;
+}
+
+class Program
+{
+    public static void F()
+    {
+    }
+}";
+            var comp = CreateCompilation(SyntaxFactory.ParseSyntaxTree(source, options: TestOptions.Regular11, path: "path/to/MyFile.cs", Encoding.Default), options: TestOptions.DebugDll);
+            WithRuntimeInstance(
+                comp,
+                references: null,
+                includeLocalSignatures: true,
+                includeIntrinsicAssembly: false,
+                validator: runtime =>
+                {
+                    var context = CreateMethodContext(runtime, "Program.F");
+                    var assembly = context.CompileExpressions(
+                        ImmutableArray.Create("C.X"),
+                        out var methodTokens,
+                        out var errorMessages);
+                    Assert.NotNull(assembly);
+                    Assert.True(errorMessages.IsEmpty);
+                    Assert.Equal(1, methodTokens.Length);
+                    assembly.VerifyIL(methodTokens[0], "<>x0.<>m0",
```

Comment:
>assembly.VerifyIL [](http://example.com/codeflow?start=20&length=17)

If we use `CompilationTestData`, would the IL have symbol names? (See previous test for instance.)
```csharp
var testData = new CompilationTestData();
var result = context.CompileExpression("z ?? x", out error, testData);
Assert.NotNull(result.Assembly);
Assert.Null(error);
testData.GetMethodData("<>x.<>m0").VerifyIL("...");
```